pkg/api/validation/validation.go

```diff
@@ -1614,6 +1615,17 @@ func ValidateVolumeMounts(mounts []api.VolumeMount, volumes sets.String, fldPath
 		if mountpoints.Has(mnt.MountPath) {
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("mountPath"), mnt.MountPath, "must be unique"))
 		}
+		for _, apath := range mountpoints.List() {
+			rel, err := filepath.Rel(mnt.MountPath, apath)
+			if err != nil {
+				// returns error if target path can't be made relative to basepath, which means no conflict
+				continue
+			}
+			if !strings.Contains(rel, "../") {
+				errorString := fmt.Sprintf("error with overlapping path:'%v'", apath)
```

Comment:
The error string here needs to be more informative, something like "volumeMounts may not overlap one another; volumeMount %v overlaps with volumeMount %v"